reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/Concat.kt

```diff
@@ -1,16 +1,20 @@
 package com.badoo.reaktive.completable
 
+import com.badoo.reaktive.base.subscribeSafe
 import com.badoo.reaktive.disposable.Disposable
 import com.badoo.reaktive.disposable.DisposableWrapper
-import com.badoo.reaktive.base.subscribeSafe
+import com.badoo.reaktive.utils.atomicreference.AtomicReference
+import com.badoo.reaktive.utils.atomicreference.getAndUpdate
 
 fun Iterable<Completable>.concat(): Completable =
     completableUnsafe { observer ->
         val disposableWrapper = DisposableWrapper()
         observer.onSubscribe(disposableWrapper)
 
-        val iterator = iterator()
-        if (!iterator.hasNext()) {
+        val index = AtomicReference(0)
+        val list = toList()
```

Comment:
Please remove `freeze()` call from `Observable.concat`, looks like it was added by mistake